smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs

```diff
@@ -93,28 +91,19 @@ pub fn execute_update_range_ticks(
 ) -> Result<Response, ContractError> {
     assert_range_admin(deps.storage, &info.sender)?;
 
-    // todo: prevent re-entrancy by checking if we have anything in MODIFY_RANGE_STATE (redundant check but whatever)
+    // prevent re-entrancy by checking if we have anything in MODIFY_RANGE_STATE
+    if MODIFY_RANGE_STATE.may_load(deps.storage)?.is_some() {
```

Comment:
how can we ever get into that situation?

Reply:
we cant, im about to deprecate this. but this was a TODO there since a while. I guess the initial intention was something like the openzeppeling reentrancyGuard https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/ReentrancyGuard.sol

Reply:
I see. Imo, reentrancy guards in cosmwasm contracts are a sign that one does not really know what's going on. There is no re-entrancy risk and caches that are consumed in replies are under our full control, so should always be empty after a tx completed.